Zend/zend_ast.c

```diff
@@ -441,3 +444,1011 @@ ZEND_API void zend_ast_apply(zend_ast *ast, zend_ast_apply_func fn) {
+			smart_str_appendc(str, ')');
+			break;
+		case ZEND_AST_INSTANCEOF: BINARY_OP(" instanceof ", 230, 231, 231);
+		case ZEND_AST_YIELD:
+			if (priority > 70) smart_str_appendc(str, '(');
+			smart_str_appends(str, "yield ");
+			if (ast->child[0]) {
+				if (ast->child[1]) {
+					zend_ast_export_ex(str, ast->child[1], 70, indent);
+					smart_str_appends(str, " => ");
+				}
+				zend_ast_export_ex(str, ast->child[0], 70, indent);
+			}
+			if (priority > 70) smart_str_appendc(str, ')');
+			break;
+		case ZEND_AST_COALESCE: BINARY_OP(" ?? ", 110, 111, 110);
+		case ZEND_AST_STATIC:
+			smart_str_appends(str, "static $");
+			zend_ast_export_name(str, ast->child[0], 0, indent);
+			APPEND_DEFAULT_VALUE(1);
+		case ZEND_AST_WHILE:
+			smart_str_appends(str, "while (");
+			zend_ast_export_ex(str, ast->child[0], 0, indent);
+			smart_str_appends(str, ") {\n");
+			zend_ast_export_stmt(str, ast->child[1], indent + 1);
+			zend_ast_export_indent(str, indent);
+			smart_str_appendc(str, '}');
+			break;
+		case ZEND_AST_DO_WHILE:
+			smart_str_appends(str, "do {\n");
+			zend_ast_export_stmt(str, ast->child[0], indent + 1);
+			zend_ast_export_indent(str, indent);
+			smart_str_appends(str, "} while (");
+			zend_ast_export_ex(str, ast->child[1], 0, indent);
+			smart_str_appendc(str, ')');
+			break;
+		case ZEND_AST_IF_ELEM:
+			if (ast->child[0]) {
+				smart_str_appends(str, "if (");
+				zend_ast_export_ex(str, ast->child[0], 0, indent);
+				smart_str_appends(str, ") ");
+			} else {
+				smart_str_appends(str, " else ");
+			}
+			if (ast->child[1]->kind == ZEND_AST_IF) {
+				zend_ast_export_list(str, (zend_ast_list*)ast->child[1], 0, 0, indent);
+			} else {
+				smart_str_appends(str, "{\n");
+				zend_ast_export_stmt(str, ast->child[1], indent + 1);
+				zend_ast_export_indent(str, indent);
+				smart_str_appendc(str, '}');
+				break;
+			}
+			break;
+		case ZEND_AST_SWITCH:
+			smart_str_appends(str, "switch (");
+			zend_ast_export_ex(str, ast->child[0], 0, indent);
+			smart_str_appends(str, ") {\n");
+			zend_ast_export_ex(str, ast->child[1], 0, indent + 1);
+			zend_ast_export_indent(str, indent);
+			smart_str_appendc(str, '}');
+			break;
+		case ZEND_AST_SWITCH_CASE:
+			zend_ast_export_indent(str, indent);
+			if (ast->child[0]) {
+				smart_str_appends(str, "case ");
+				zend_ast_export_ex(str, ast->child[0], 0, indent);
+				smart_str_appends(str, ":\n");
+			} else {
+				smart_str_appends(str, "default:\n");
+			}
+			zend_ast_export_stmt(str, ast->child[1], indent + 1);
+			break;
+		case ZEND_AST_DECLARE:
+			smart_str_appends(str, "declare(");
+			if (ast->child[0]->kind == ZEND_AST_CONST_DECL) {
+				zend_ast_export_list(str, (zend_ast_list*)ast->child[0], 1, 0, indent);
+			} else {
+				zend_ast_export_ex(str, ast->child[0], 0, indent);
+			}
+			if (ast->child[1]) {
+				smart_str_appends(str, " = ");
+				zend_ast_export_ex(str, ast->child[1], 0, indent);
+			}
+			smart_str_appendc(str, ')');
```

Comment:
Hm, I don't think I get the `declare` code at all ... shouldn't `ast->child[0]` be the list of key-value pairs and `ast->child[1]` the code block?